### PR TITLE
convert oauth test to functional test

### PR DESCRIPTION
**Pull request by McKnight-42**
resolves #

### Description
`dbt-snowflake` integration test conversion of `oauth` test to new functional format
### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Review by stu-k — APPROVED** [review]

File: tests/functional/oauth/test_oauth.py [hunk 1]
```diff
@@ -75,7 +96,6 @@ def snowflake_profile(self):
         return profile
 
     @pytest.mark.skipif(OAUTH_TESTS_DISABLED, reason='oauth tests disabled')
-    @use_profile('snowflake')
-    def test_snowflake_basic(self):
-        self.run_dbt()
-        self.assertManyRelationsEqual([['MODEL_3'], ['MODEL_4']])
+    def test_snowflake_basic(self, project):
```
Comment (McKnight-42) [inline tests/functional/oauth/test_oauth.py:99]: QUESTION: Was trying to find a way more inline with new functional test to get this setup cutdown, using `profiles_config_update` but didn't receive any real luck or should we directly override the `dbt_profile_target`

Comment (VersusFacit) [reply tests/functional/oauth/test_oauth.py:101]: profiles_config_update is for dbt_project yml, but you're handing it target params. Use the target fixture and this should be alright.

**Review by VersusFacit — CHANGES_REQUESTED** [review]
Left some comments (apparently I never hit full request? The UI is ambiguous but I hope my comments have been visible to you)

File: tests/functional/oauth/test_oauth.py (hunk 1 above)
Comment (McKnight-42) [reply tests/functional/oauth/test_oauth.py:101]: will play around with that method again was trying to either trigger it via that but hit errors so tried the config update, or using a new conftest option which felt bloated as we wouldn't use it anywhere else.

File: tests/functional/oauth/test_oauth.py [hunk 2]
```diff
@@ -30,10 +30,34 @@
 Update CI providers and test.env with the new values (If you kept the security
 integration the same, just the refresh token changed)
 """
-import os
+
 import pytest
-from tests.integration.base import DBTIntegrationTest, use_profile
+import os
+from dbt.tests.util import run_dbt, check_relations_equal
```
[the remark this answers is not in the document]
Comment (McKnight-42) [reply tests/functional/oauth/test_oauth.py:36]: fix

File: tests/functional/oauth/test_oauth.py [hunk 3]
```diff
@@ -48,18 +72,15 @@ def env_set_truthy(key):
 OAUTH_TESTS_DISABLED = env_set_truthy('DBT_INTEGRATION_TEST_SNOWFLAKE_OAUTH_DISABLED')
```
[the remark this answers is not in the document]
Comment (McKnight-42) [reply tests/functional/oauth/test_oauth.py:72]: no even in the integration test doesn't seem like it was actually called, might of originally been treated elsewhere

File: tests/functional/oauth/test_oauth.py [hunk 4]
```diff
@@ -48,18 +72,15 @@ def env_set_truthy(key):
...
     def models(self):
-        return self.dir("models")
+        return {
+            "model_1.sql": _MODELS__MODEL_1_SQL,
+            "model_2.sql": _MODELS__MODEL_2_SQL,
+            "model_3.sql": _MODELS__MODEL_3_SQL,
+            "model_4.sql": _MODELS__MODEL_4_SQL,
+            }
 
     def snowflake_profile(self):
```
[the remark this answers is not in the document]
Comment (McKnight-42) [reply tests/functional/oauth/test_oauth.py]: not sure seems to of been lost along the way even integration test doesn't have it called will walk back in history as much as I can and see if i can return with an answer

File: tests/functional/oauth/test_oauth.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "model_1.sql": _MODELS__MODEL_1_SQL,
+            "model_2.sql": _MODELS__MODEL_2_SQL,
+            "model_3.sql": _MODELS__MODEL_3_SQL,
+            "model_4.sql": _MODELS__MODEL_4_SQL,
+            }
```
Comment (VersusFacit) [inline tests/functional/oauth/test_oauth.py:85]: nit: formatting

**Review by VersusFacit — APPROVED** [review]
One leftover nit but we're looking good!

File: tests/functional/oauth/test_oauth.py (hunk 5 above)
Comment (McKnight-42) [reply tests/functional/oauth/test_oauth.py:85]: @VersusFacit what is the specific formatting issue here for clarification?

Comment (McKnight-42) [reply tests/functional/oauth/test_oauth.py]: oh see it, good catch

**Pull request merged**